MyLibs/fishAnalysis.cpp

```diff
@@ -432,3 +432,164 @@ void rot90CW(Mat src, Mat dst)
+		fishCenterPoint.push_back(fishCenter);
+
+		//Find the head of fish
+		vector<Point> boundingBox_midpoint(2);
+		findMidpoint(fishContours[0], boundingBox_midpoint, true);
+		vector<Point> fishHeadContour;
+		vector<int> indices = findPtsLineIntersectContour(fishContours[0], boundingBox_midpoint[0], boundingBox_midpoint[1]);
+		vector<vector<Point>> contourHalves(2);
+
+		contourHalves[0].insert(contourHalves[0].end(), fishContours[0].begin() + indices[0], fishContours[0].begin() + indices[1]);
+		contourHalves[1].insert(contourHalves[1].end(), fishContours[0].begin() + indices[1], fishContours[0].end());
+		contourHalves[1].insert(contourHalves[1].end(), fishContours[0].begin(), fishContours[0].begin() + indices[0]);
+
+		vector<double> areas(2);
+		if (contourHalves[0].size()*contourHalves[1].size() == 0) {
+			cout << "Can't find the headcontour!" << endl;
+			return false;
+		}
+
+		areas[0] = contourArea(contourHalves[0]);
+		areas[1] = contourArea(contourHalves[1]);
+		if (areas[0] > areas[1]) {
+			fishHeadContour = contourHalves[0];
+		}
+		else {
+			fishHeadContour = contourHalves[1];
+		}
+
+
+		Point fishHead;
+		vector<Point> headBoundingBox_midpoint(2);
+		findMidpoint(fishHeadContour, headBoundingBox_midpoint, false);
+		if (norm(fishCenter - headBoundingBox_midpoint[0]) < norm(fishCenter - headBoundingBox_midpoint[1]))
+			fishHeadPoint.push_back(headBoundingBox_midpoint[1]);
+		else
+			fishHeadPoint.push_back(headBoundingBox_midpoint[0]);
+		return false;
+	}
+	else {
+
+
+		vector<Point> fishCenterPoint_reserve, fishHeadPoint_reserve;
+		Point fishCenter_mean = findCenter(fishCenterPoint), fishHead_mean = findCenter(fishHeadPoint);
+
+		for (int i = 0; i < 50; i++) {
+			if (norm(fishCenterPoint[i] - fishCenter_mean) < 6)
+				fishCenterPoint_reserve.push_back(fishCenterPoint[i]);
+			if (norm(fishHeadPoint[i] - fishHead_mean) < 6)
+				fishHeadPoint_reserve.push_back(fishHeadPoint[i]);
+		}
+		if (fishCenterPoint_reserve.size() < 30) {
+			cout << "Can't find the Center!Reserve point:" << fishCenterPoint_reserve.size() << endl;
+
+		}
+		if (fishHeadPoint_reserve.size() < 30) {
+			cout << "Can't find the head!Reserve point:" << fishHeadPoint_reserve.size() << endl;
+
+		}
+		fishCenterPoint[0] = findCenter(fishCenterPoint_reserve);
+		fishHeadPoint[0] = findCenter(fishHeadPoint_reserve);
+		cout << "Center and Head have been found." << endl;
+		return true;
+	}
+}
+
+/* This function return a radian to describe the fishtailing motion */
+bool fishAngleAnalysis(Mat fishImg, Point fishHead, Point fishCenter, Point* fishTail_return,double *fishAngle) {
+	//Find the contour of fish
+	Mat binaryzation, closeImg;
+	vector<vector<Point>> allContours, fishContours;
+	threshold(fishImg, binaryzation, 10, 255, CV_THRESH_BINARY);
+	findContours(binaryzation, allContours, CV_RETR_LIST, CHAIN_APPROX_NONE);
+	for (int i = 0; i < allContours.size(); i++) {
+		if (contourArea(allContours[i]) < 700 && contourArea(allContours[i]) > 500)
```

Comment:
Better use variables to indicate the meaning of these two variables

Reply:
Like this?
```
double threshold_val = 10, max_val = 255;
threshold(fishImg, binaryzation, threshold_val, max_val, CV_THRESH_BINARY);
```